### PR TITLE
SW-483 added tier switcher for octoprint

### PR DESCRIPTION
**Pull request by JFSolution**
octoprint will switch between prerelease (alpha or beta branch) and release in combination of the tier selected

### CODE REVIEW

File: octoprint_mrbeam/software_update_information.py [hunk 1]
```diff
@@ -89,16 +89,26 @@ def switch_software_channel(plugin, channel):
...
-#         )
-#     )
+def _config_octoprint(plugin, tier):
+    prerelease_channel = None
+    type = "github_release"
+    if tier in [SW_UPDATE_TIER_ALPHA, SW_UPDATE_TIER_BETA]:
+        prerelease_channel = "mrbeam2-{tier}"
+
+    if tier in [SW_UPDATE_TIER_DEV]:
+        type = "github_commit"
```
Comment (amrsoll) [inline octoprint_mrbeam/software_update_information.py]: 1. Please use `elif`
2. Either change the acceptance criteria in the ticket or change `type` for develop to become prerelease-based and use `develop` as the `prerelease_channel`

**Review by amrsoll — CHANGES_REQUESTED** [review]

File: octoprint_mrbeam/software_update_information.py (hunk 1 above)
Comment (JFSolution) [reply octoprint_mrbeam/software_update_information.py]: 1. ohh good point
2. I changed the acceptace criteria, I makes more sense to track the github commits for the develop tier

**Review by amrsoll — DISMISSED** [review]

File: octoprint_mrbeam/software_update_information.py [hunk 2]
```diff
@@ -309,20 +307,18 @@ def _get_package_description(
         branch_default = branch_default.format(tier=get_tier_by_id(tier))
     if prerelease_channel and "{tier}" in prerelease_channel:
         prerelease_channel = prerelease_channel.format(tier=get_tier_by_id(tier))
+        kwargs.update(prerelease_channel=prerelease_channel)
```
Comment (amrsoll) [inline octoprint_mrbeam/software_update_information.py]: I see.. We should probably remove all the options that have a `None` value before passing it onto OctoPrint through the hook?

also you can shorten this to 1 line

**Review by amrsoll — DISMISSED** [review]

**Review by amrsoll — APPROVED** [review]
